shared/devices/index.render.desktop.js

```diff
@@ -1,160 +1,193 @@
-      </div>
+      <Box>
+        <Box style={stylesRevokedRow} onClick={this.toggleHeader}>
+          <Text type='BodySemibold'>Revoked devices</Text>
+          <Icon type={iconType} style={{padding: 5}}/>
+        </Box>
+        {this.state.expanded && this.props.children}
+      </Box>
     )
   }
+}
+
+class DeviceRow extends Component {
+  state: {
+    highlighted: boolean
+  };
+  props: {
+    device: DeviceType,
+    revoked?: boolean,
+    showRemoveDevicePage?: () => void,
+    showExistingDevicePage?: () => void
+  };
 
-  renderDevice (device) {
-    if (device.type === 'computer') {
-      return this.renderDesktop(device)
-    } else if (device.type === 'mobile') {
-      return this.renderPhone(device)
-    } else if (device.type === 'paper key') {
-      return this.renderPaperKey(device)
-    } else {
-      console.warn('Unknown device type: ' + device.type)
+  constructor (props) {
+    super(props)
+    this.state = {
+      highlighted: false
     }
+    this.highlightOn = this.highlightOn.bind(this)
+    this.highlightOff = this.highlightOff.bind(this)
+  }
+
+  highlightOn (e) {
+    this.setState({highlighted: true})
+  }
+
+  highlightOff (e) {
+    this.setState({highlighted: false})
   }
 
   render () {
+    const icon: IconProps.type = {
+      'mobile': 'phone-bw-m',
+      'desktop': 'computer-bw-s-2',
+      'backup': 'paper-key-m'
+    }[this.props.device.type]
+
+    let textStyle = {fontStyle: 'italic'}
+    if (this.props.revoked) {
+      textStyle = {
+        ...textStyle,
+        color: globalColors.black40,
+        textDecoration: 'line-through'
+      }
+    }
+
     return (
-      <div>
-        <div style={styles.deviceContainer}>
-          <div style={{...styles.deviceOuter, ...styles.deviceAction}} onClick={() => this.props.showExistingDevicePage()}>
-            <div style={styles.device}>
-              <ActionNoteAdd style={styles.deviceIcon} />
-              <h3>Connect a new device</h3>
-              <p style={{...styles.line4, ...styles.actionDesc}}>On another device, download Keybase then click here to enter your unique passphrase.</p>
-            </div>
-          </div>
-
-          <div style={{...styles.deviceOuter, ...styles.deviceAction}} onClick={() => this.props.showGenPaperKeyPage()}>
-            <div style={styles.device}>
-              <CommunicationVpnKey style={styles.deviceIcon} />
-              <h3>Generate a new paper key</h3>
-              <p style={{...styles.line4, ...styles.actionDesc}}>Portland Bushwick mumblecore.</p>
-            </div>
-          </div>
-        </div>
-
-        <div style={styles.deviceContainer}>
-          {this.props.devices && this.props.devices.map(device => this.renderDevice(device))}
-        </div>
-      </div>
+      <Box key={this.props.device.name} onMouseOver={this.highlightOn} onMouseOut={this.highlightOff} style={{...stylesCommonRow, backgroundColor: this.props.revoked ? globalColors.lightGrey : globalColors.white}}>
```

Comment:
previously we've been using css pseudo selectors to do this kind of thing. this approach is fine, just pointing it out